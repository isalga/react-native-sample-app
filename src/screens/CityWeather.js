// @flow
import type { City, Weather } from '../entities/types'
import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { connect } from 'react-redux'
import { getSelectedCity } from '../entities/city/selectors'
import { getWeatherByCityId } from '../entities/weather/selectors'
import WeatherPanel from '../entities/weather/components/Panel'
import { colors } from '../styles'

type Props = {
  city: ?City,
  weather: ?Weather
}

export const CityWeather = ({ weather, city }: Props) =>
  <View style={styles.container}>
    <WeatherPanel weather={weather} title={city && city.name}/>
  </View>

CityWeather.navigationOptions = {
  title: 'Weather Info'
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    backgroundColor: colors.background,
  }
})

export default connect(
  (state) => ({
    city: getSelectedCity(state),
    weather: getWeatherByCityId(state)(state.selectedCityId)
  }),
  (dispatch) => ({})
)(CityWeather)
