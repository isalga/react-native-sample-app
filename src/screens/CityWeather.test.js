import React from 'react'
import ReactDOM from 'react-dom'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { CityWeather } from './CityWeather'

Enzyme.configure({ adapter: new Adapter() })

const city = {
  id: '1',
  displayName: 'Abu Dhabi',
  name: 'Abu_Dhabi',
  country: 'united_arab_Emirates'
}

const weather = {
  cityId: '1',
  temperature: 'testtemperature',
  windDirection: 'testwindDirection',
  windSpeed: 'testwindSpeed',
  humidity: 'testhumidity',
  precipitation: 'testprecipitation'
}

describe('CityWeather screen component', () => {

  it('should render correctly', () => {
    const wrapper = Enzyme.shallow(
      <CityWeather
        city={city}
        weather={weather}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
