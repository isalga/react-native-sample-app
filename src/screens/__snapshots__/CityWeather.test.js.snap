// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityWeather screen component should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "ghostwhite",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <WeatherPanel
    title="Abu_Dhabi"
    weather={
      Object {
        "cityId": "1",
        "humidity": "testhumidity",
        "precipitation": "testprecipitation",
        "temperature": "testtemperature",
        "windDirection": "testwindDirection",
        "windSpeed": "testwindSpeed",
      }
    }
  />
</View>
`;
